### DISCUSSION TITLE
Valgrind en Docker

### DISCUSSION THREAD

**Discussion by trinisch** [IIC2133-PUC/2024-1#114] [Ayuda] [answered]
Hola! Estaba tratando de correr valgrind en docker y para eso cloné el repositorio edd-docker. Para que funcionara puse el comando "make start" y luego probé el comando valgrind ./program en la carpeta "probar_setup" y funcionó bien. Después copié mi carpeta de la tarea 1 en volume, pero cuando ejecuto el comando valgrind ./market o valgrind ./hsearch, aparece el error: valgrind: ./market (o ./hsearch): cannot execute binary file. Saben como solucionar esto?

**Comment by benjavicente** [comment, marked as answer]
Probablemente, es porque compilaste el programa en un sistema operativo (windows o mac), y luego lo metiste a otro (linux). Corre

```sh
make clean && make
```

y ve si te funciona valgrind

**Reply by trinisch** [reply]
Ahí funcionó! Muchas gracias :)
